containers/src/Data/IntMap/Internal.hs

```diff
@@ -598,7 +599,23 @@ notMember k m = not $ member k m
 lookup :: Key -> IntMap a -> Maybe a
 lookup !k = go
   where
-    go (Bin p m l r) | nomatch k p m = Nothing
+    go (Bin _p m l r)
+                     | zero k m  = go l
+                     | otherwise = go r
+    go (Tip kx x) | k == kx   = Just x
+                  | otherwise = Nothing
+    go Nil = Nothing
+
+-- | /O(min(n,W))/. 'query' has identical behaviour to 'Data.IntMap.Internal.lookup', but
+-- will fail faster in the case that the key is not present. It is otherwise
+-- preferable to use `Data.IntMap.Internal.lookup'.
```

Comment:
```suggestion
-- | /O(min(n,W))/. 'query' has identical behaviour to 'Data.IntMap.Internal.lookup', but
-- will fail faster if the key is not present. When the key is (likely) present, it is
-- preferable to use 'Data.IntMap.Internal.lookup' instead.
```
This is clearer IMO.

Reply:
Ok, it is worth saying that is is more subtle in reality than the key being present or not present and depends instead on whether the key shares a bit prefix with the current tree keys but that subtlety is hard to capture.